Add an is_tuple_like trait for sequenceable type detection

Naios opened the pull request:
Requires #2685, otherwise the unit test won't pass

hkaiser commented on hpx/traits/is_tuple_like.hpp:
```diff
@@ -0,0 +1,27 @@
+        util::tuple_size<T>::value
+    )>::type> : std::true_type { };
+}}
+
```
Please try to adapt your style of formatting to the rest of HPX. I'll send you a .clang-format file which you can use as a starting point.

hkaiser commented:
Please add the new #include here: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/include/traits.hpp

Naios commented:
I fixed the code formatting and added the trait to the include compilation.
@hkaiser thank you for sending me the `.clang-format` file, it seems to be much better than the one I had.

hkaiser commented:
@Naios: the current build failures are caused by a bogus commit you picked up by merging from master. Please merge again, this should resolve the issue.

Naios commented:
@hkaiser I rebased the PR accordingly, but the CI is still failing
